File: C++/Swap-Nodes-in-Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if (head == NULL || head->next == NULL) return head;
        ListNode *dummy = new ListNode(0), *prev = dummy;
        while (head != NULL && head->next != NULL) {
            ListNode *temp1 = head, *temp2 = head->next;
            head = head->next->next;
            prev->next = temp2;
            prev = prev->next;
            prev->next = temp1;
            prev = prev->next;
        }
        prev->next = head;
        return dummy->next;
    }
};

File: Python/Swap-Nodes-in-Pairs.py
#!/usr/bin/ERNV python

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param a ListNode
    # @return a ListNode
    def swapPairs(self, head):
        if head == None or head.next == None:
            return head
        node = head.next.next
        root = head.next
        root.next = head
        tail = head
        while node != None:
            if node.next != None:
                node1 = node
                node2 = node.next
                node = node.next.next
                tail.next = node2
                node2.next = node1
                tail = node1
            else:
                tail.next = node
                tail = node
                break
        tail.next = None
        return root